### PR TITLE
feat(ld-notice): add success mode and custom icon

### PR DESCRIPTION
**Pull request by Tabanion**
# Description

Added 'success' mode for Notice component.
Added possibility to display custom icons.

# How Has This Been Tested?

Added relevant unit/e2e tests, snapshot and screenshots.

- [ ✓ ] unit tests
- [ ✓ ] e2e tests
- [ ✓ ] accessibility tests
- [ ✓ ] tested manually

**Test Configuration**:

- liquid versions:
- os:
- browsers:
- screen readers:
- misc:

# Checklist:

- [ ✓ ] My code follows the style guidelines of this project
- [ ✓ ] I have performed a self-review of my own code
- [ ✓ ] I have commented my code, particularly in hard-to-understand areas
- [ ✓ ] I have made corresponding changes to the documentation
- [ ✓ ] My changes generate no new warnings
- [ ✓ ] I have added tests that prove my fix is effective or that my feature works
- [ ✓ ] New and existing tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-notice/test/ld-notice.spec.tsx [hunk 1]
```diff
@@ -28,4 +28,17 @@ describe('ld-notice', () => {
     })
     expect(page.root).toMatchSnapshot()
   })
+  fit('renders with custom icon', async () => {
```
Comment (borisdiakur) [inline src/liquid/components/ld-notice/test/ld-notice.spec.tsx:31]:
```suggestion
  it('renders with custom icon', async () => {
```

File: src/liquid/components/ld-notice/test/ld-notice.spec.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { h } from '@stencil/core'
 import { newSpecPage } from '@stencil/core/testing'
 import { LdNotice } from '../ld-notice'
 
-describe('ld-notice', () => {
+fdescribe('ld-notice', () => {
```
Comment (borisdiakur) [inline src/liquid/components/ld-notice/test/ld-notice.spec.tsx:5]:
```suggestion
describe('ld-notice', () => {
```

File: src/liquid/components/ld-notice/ld-notice.tsx [hunk 3]
```diff
@@ -8,21 +9,41 @@ import { Component, h, Prop, Host } from '@stencil/core'
...
   @Prop() headline?: string
 
   /** Mode of the notice. */
-  @Prop() mode: 'error' | 'info' | 'warning' = 'info'
+  @Prop() mode: 'error' | 'info' | 'warning' | 'success' = 'info'
+
+  @State() hasCustomIcon: boolean
+
+  componentWillLoad() {
+    this.hasCustomIcon = !!this.hostElement.querySelector(
```
Comment (borisdiakur) [inline src/liquid/components/ld-notice/ld-notice.tsx:29]: This should be a State variable, see https://stenciljs.com/docs/state#state

File: src/liquid/components/ld-notice/ld-notice.tsx [hunk 4]
```diff
@@ -8,21 +9,41 @@ import { Component, h, Prop, Host } from '@stencil/core'
  * @part icon - Image tag used for the icon
  */
 @Component({
+  assetsDirs: ['assets'],
```
Comment (borisdiakur) [inline src/liquid/components/ld-notice/ld-notice.tsx:12]:
```suggestion
```

**Review by borisdiakur — CHANGES_REQUESTED** [review]
The screenshot [screenshot/images/4eecc499b8ec45721541f5fc13f73956.png](https://github.com/emdgroup-liquid/liquid/pull/316/files?short_path=47fbe60#diff-47fbe604ee0ef77a8cb376513ce799bbfb4c4ed58fa61a9d22593e5d97641ac2) looks weird, can you check what's wrong there?

File: src/liquid/components/ld-notice/test/ld-notice.e2e.ts [hunk 5]
```diff
@@ -187,6 +239,36 @@ describe('ld-notice', () => {
...
+          </div>`,
+          { components: [LdIcon, LdNotice] }
+        )
+        const results = await page.compareScreenshot()
+        expect(results).toMatchScreenshot()
+      })
+    })
+
+    describe('with custom icon', () => {
+      it('custom icon', async () => {
```
Comment (renet) [inline src/liquid/components/ld-notice/test/ld-notice.e2e.ts:261]: see above

**Review by renet — CHANGES_REQUESTED** [review]
.

File: src/liquid/components/ld-notice/ld-notice.tsx (hunk 4 above)
Comment (Tabanion) [reply src/liquid/components/ld-notice/ld-notice.tsx:12]: isn't this the way to point to the assets directory?

Comment (borisdiakur) [reply src/liquid/components/ld-notice/ld-notice.tsx:12]: I just haven't seen that you are using any asset. You should only need it if you are using the [`getassetpath`](https://stenciljs.com/docs/assets#getassetpath) function.

**Pull request closed without merging**